ui/src/dashboards/selectors/index.ts

```diff
@@ -15,6 +15,47 @@ export const getTimeRange = (state: AppState): TimeRange => {
   return state.ranges[contextID] || DEFAULT_TIME_RANGE
 }
 
+export const getTimeRangeWithTimezone = (state: AppState): TimeRange => {
+  const timeRange = getTimeRange(state)
+  const timeZone = getTimeZone(state)
+  // create a copy of the timeRange so as not to mutate the original timeRange
+  const newTimeRange = Object.assign({}, timeRange)
```

Comment:
```suggestion
  const newTimeRange = {...timeRange}
```
i *believe* we prefer es6 notation to `Object.assign`

you can leave the comment if you like, but i think seeing `Object.assign({}, foo)` or `{...foo}` generally implies immutable copying which makes the comment a little redundant.